boards/nrf52840dongle/board.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2019 Christian Amsüss <chrysn@fsfe.org>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_nrf52840dongle
+ * @{
+ *
+ * @file
+ * @brief       Board initialization for the nRF52840-dongle
+ *
+ * @author      Christian Amsüss <chrysn@fsfe.org>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "board.h"
+
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* initialize the board's single LED */
+    gpio_init(LED0_PIN, GPIO_OUT);
+    gpio_set(LED0_PIN);
+
+    /* initialize the board's RGB LED */
+    gpio_init(LED1_PIN, GPIO_OUT);
+    gpio_set(LED1_PIN);
+    gpio_init(LED2_PIN, GPIO_OUT);
+    gpio_set(LED2_PIN);
+    gpio_init(LED3_PIN, GPIO_OUT);
+    gpio_set(LED3_PIN);
+
+    /* initialize the board's button */
+    gpio_init(BTN0_PIN, BTN0_MODE);
+    gpio_set(BTN0_PIN);
```

Comment:
No need to init the button. This will be done by the application. `gpio_set()` also makes no sense here.

Reply:
At least it's automatic in SAUL (but it may be a good idea to rid all other boards of this then, I've seen at least nrf51dk do the very same).

Removed with next batch.